msticpy/data/uploaders/splunk_uploader.py

```diff
@@ -0,0 +1,252 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Splunk Uploader class."""
+from pathlib import Path
+from typing import Any
+from tqdm.notebook import tqdm
+import pandas as pd
+from pandas.io.parsers import ParserError
+
+from .uploader_base import UploaderBase
+from ..._version import VERSION
+from ..drivers.splunk_driver import SplunkDriver
+from ...common.exceptions import MsticpyConnectionError, MsticpyUserError
+
+__version__ = VERSION
+__author__ = "Pete Bryan"
+
+
+class SplunkUploader(UploaderBase):
+    """Uploader class for Splunk."""
+
+    def __init__(self, username: str, host: str, password: str, **kwargs):
+        """Initialize a Splunk Uploader instance."""
+        super().__init__()
+        self._kwargs = kwargs
+        self.workspace = host
+        self.workspace_secret = password
+        self.user = username
+        self.driver = SplunkDriver()
+        self.port = kwargs.get("port", 8089)
+        self._debug = kwargs.get("debug", False)
+        self._connect = kwargs.get("connect", True)
+        self.connected = False
+        if self._connect:
+            self.connect()
+
+    def connect(self):
+        """Connect to Splunk host."""
+        self.driver.connect(
+            host=self.workspace,
+            username=self.user,
+            password=self.workspace_secret,
+            port=self.port,
+        )
+        self.connected = True
+
+    def _post_data(
+        self,
+        data: pd.DataFrame,
+        index_name: str,
+        table_name: Any,
+        host: str = None,
+        **kwargs,
+    ):
+        """
+        Write data to the Splunk instance connected to.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            Data to upload.
+        index_name : str
+            Name of the Splunk Index to add data to, if it doesn't exist it will be created.
+        table_name : str
+            The souretype in Splunk data will be uploaded to.
+        host : str, optional
+            The hostname associated with the uploaded data, by default "Upload".
+
+        """
+        if not self.connected:
+            raise MsticpyConnectionError(
+                "Splunk host not connected, please call .connect before proceding.",
+                title="Splunk host not connected",
+            )
+        if not host:
+            host = "Upload"
+        create_idx = kwargs.get("create_index", "False")
```

Comment:
Maybe make this an explicit param since it affects a critical element of the operation. 
Actually this and previous comment really apply to public methods rather than this one.